### PR TITLE
fix(ivy): recompile on template change in ngc watch mode on Windows

### LINKED ISSUE #32869: ngc --watch ignores changes in templates (ivy, 9.0.0-next.8)
This is a repost of #31982, since:
- the bug is still there in 9.0.0-next.8 and 8.2.8 , 
- it is reproducible (as before) with the given example and 
- it has been closed unjustified as duplicate by @alxhub.

🐞 **bug report**
**Affected Package**
The issue is caused by package @angular/compiler-cli

**Is this a regression?**
Yes, the previous version in which this bug was not present was: **8.1.2**
(But, due to #31322, it only worked on every second change of a template.)

**Description**
I run ngc in watch mode. Output directory is out-tsc.
Ivy is enabled.
I change a component's template.
ngc says "File change detected. Starting incremental compilation."
However, the template change does not result in an according change in the *.js file in out-tsc directory.
For changes in *.ts files, it works.

🔬 **Minimal Reproduction**
Example project is at: https://github.com/stefanocke/angular-cli-app-with-rollup
Please run:

npm i
npm run ngc-app-watch

(This is nothing more than running ngc in watch mode, as the title of this bug says.)

Then, change **app.component.html** and look for **app.component.js** in tsc-out.

You will see that, although the console says correctly "File change detected. Starting incremental compilation.", nothing has changed in app.component.js.

🌍 **Your Environment**
Angular Version:

Angular CLI: 8.3.6
Node: 10.16.0
OS: win32 x64
Angular: 9.0.0-next.8
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-next.6
@angular-devkit/build-angular     0.900.0-next.6
@angular-devkit/build-optimizer   0.900.0-next.6
@angular-devkit/build-webpack     0.900.0-next.6
@angular-devkit/core              9.0.0-next.6
@angular-devkit/schematics        8.3.6
@angular/cli                      8.3.6
@ngtools/webpack                  9.0.0-next.6
@schematics/angular               8.3.6
@schematics/update                0.803.6
rxjs                              6.5.2
typescript                        3.5.3

**Anything else relevant?**
Ivy is enabled

### PR DESCRIPTION
**Pull request by JoostK**
In #33551, a bug in `ngc --watch` mode was fixed so that a component is
recompiled when its template file is changed. Due to insufficient
normalization of files paths, this fix did not have the desired effect
on Windows.

Fixes #32869

### CODE REVIEW

**Comment by JoostK** [comment]
@stefanocke Hi Stefan, could you please test the artifacts from this PR on Windows?

Artifacts: https://app.circleci.com/jobs/github/angular/angular/540987/artifacts
Instructions: https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#archives-for-each-package

I am expecting this PR to fix your issue, hoping you can confirm. Thanks!

**Comment by stefanocke** [comment]
@JoostK , sorry, but I get a 404 on https://app.circleci.com/jobs/github/angular/angular/540987/artifacts

**Comment by stefanocke** [comment]
@JoostK , it works for me. :-)

**Comment by stefanocke** [comment]
Hello @JoostK , any chance to get it merged in the next RC? Thank you.

**Comment by JoostK** [comment]
@stefanocke yeah, I can look into reviving this PR. There has been quite some work around incremental compilations and dependency tracking that conflicted heavily with this PR so it has gotten outdated.

File: packages/compiler-cli/src/perform_watch.ts [hunk 1]
```diff
@@ -246,11 +246,13 @@ export function performWatchCompilation(host: PerformWatchHost):
   }
 
   function watchedFileChanged(event: FileChangeEvent, fileName: string) {
+    const normalizedPath = path.normalize(fileName);
```
Comment (petebacondarwin) [inline packages/compiler-cli/src/perform_watch.ts]: Why the changes in this file? It looks like a separate bit of tidying up that are not relevant to this commit, no?
If it is necessary for this fix, then should we not be using a platform independent normalization of the path?

**Review by petebacondarwin — APPROVED** [review]
One minor concern.
Thanks  @JoostK !

File: packages/compiler-cli/src/perform_watch.ts (hunk 1 above)
Comment (JoostK) [reply packages/compiler-cli/src/perform_watch.ts]: The primary change here is passing the normalized path to `startTimerForRecompilation`. While I believe (I don't recall exactly) this is indeed not strictly necessary for this to work, but it makes more sense to me to be passing the normalized path.

In any case, this code resides in "ngc" where we don't have the abstracted filesystem, or did you mean something else?

Comment (petebacondarwin) [reply packages/compiler-cli/src/perform_watch.ts]: Oh I didn't notice it was outside of ngtsc.
But you could try `path.posix.normalize`?

**Review by petebacondarwin — APPROVED** [review]

File: packages/compiler-cli/src/perform_watch.ts (hunk 1 above)
Comment (JoostK) [reply packages/compiler-cli/src/perform_watch.ts]: I think that only switching over to `path.posix.normalize` would actually also resolve this issue, even without the change to ngtsc. I made the ngtsc changes so that we can no longer go wrong in this regard, as the branded paths ensure the paths are all standardized.

Actually, come to think of it, the `CompilerHost.getModifiedResourceFiles` can be provided by third parties which may or may not use posix paths, so the ngtsc change is still valuable.

In the end I don't _think_ that changing this to `path.posix.normalize` would behave differently (apart from the paths containing forward slashes, instead of backwards slashes).

**Comment by JoostK** [comment]
merge-assistance: Pete's review should be sufficient, I am code-owner of fw-compiler.

**Review by devversion — APPROVED** [review]
LGTM for the BUILD file update (for which pullapprove requested review from dev-infra)

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:293157101:BASE:293157144:1580835701660:3a873610)
